contributors/design-proposals/support-union-in-api-server.md

```diff
@@ -0,0 +1,200 @@
+Add Tag Indicating To Replace *Union* Fields in Patch
+=============
+
+Add a tag `patchStrategy:"replaceKey"`. For a given type that has this tag, all keys/fields missing 
+from the request will be cleared when patching the object. For a field presents in the request,
+it will be merged with the live config.
+
+This approach could be used to fix the issues with `kubectl apply` (e.g. [#34292](https://github.com/kubernetes/kubernetes/issues/34292))
```

Comment:
Go into more detail about this use case for addressing unions and kubectl apply.  Start what problems this solves with patch - use case + example.

Use cases:
- As a user patching a map, I want keys mutually exclusive with those that I am providing to automatically be cleared.
- As a user running `kubectl apply`, when I update a field in my configuration file, I want mutually exclusive fields never specified in my configuration to be cleared by apply

Examples:
- General Example: Keys in a Union are mutually exclusive.  Clear unspecified union values in a Union that contains a discriminator.
- Specific Example: When patching a Deployment `.spec.strategy`, clear `.spec.strategy.rollingUpdate` if it is not provided in the patch so that changing  `.spec.strategy.type` will not fail.
- General Example: Keys in a Union are mutually exclusive.  Clear unspecified union values in a Union that does not contain a discriminator.
- Specific Example: When patching a Pod `.spec.volume`, clear all volume fields except the one specified in the patch.